pkg/api/accesscontrol.go

```diff
@@ -606,14 +606,54 @@ func (hs *HTTPServer) declareFixedRoles() error {
 		Grants: []string{"Admin"},
 	}
 
+	snapshotsCreatorRole := ac.RoleRegistration{
+		Role: ac.RoleDTO{
+			Name:        "fixed:snapshots:creator",
+			DisplayName: "Creator",
+			Description: "Create snapshots",
+			Group:       "Snapshots",
+			Permissions: []ac.Permission{
+				{Action: dashboards.ActionSnapshotsCreate},
+			},
+		},
+		Grants: []string{string(org.RoleEditor)},
+	}
+
+	snapshotsDeleterRole := ac.RoleRegistration{
+		Role: ac.RoleDTO{
+			Name:        "fixed:snapshots:deleter",
+			DisplayName: "Deleter",
+			Description: "Delete snapshots",
+			Group:       "Snapshots",
+			Permissions: []ac.Permission{
+				{Action: dashboards.ActionSnapshotsDelete},
+			},
+		},
+		Grants: []string{string(org.RoleEditor)},
+	}
+
+	snapshotsReaderRole := ac.RoleRegistration{
+		Role: ac.RoleDTO{
+			Name:        "fixed:snapshots:reader",
+			DisplayName: "Reader",
+			Description: "Read snapshots",
+			Group:       "Snapshots",
+			Permissions: []ac.Permission{
+				{Action: dashboards.ActionSnapshotsRead},
+			},
+		},
+		Grants: []string{string(org.RoleViewer)},
```

Comment:
This role is assigned by default to Viewers to preserve the current workflow, in which all viewers can list Snapshots they created.